### DISCUSSION TITLE
Expand menu APIs to Explorer -> Open Editors menu

### DISCUSSION THREAD

**Discussion by Balastrong** [microsoft/vscode-discussions#1122] [Extension Development QnA] [answered]
## Request

I'd like to add an icon button and a button in the context menu of the Open Editors menu but it seems the api is not exposed.

![image](https://github.com/microsoft/vscode-discussions/assets/7253929/21255920-f9d3-4cb9-a49d-6fdfea8e7f28)

For example, I see that I can add buttons in the Source Control -> Source Control bar by adding an entry on my package.json like
```
    "menus": {
      "scm/title": [
        {
          "command": "my-command",
        }
      ]
    },
```
I can also add an entry on the context menu of files in the explorer tab with:
```
    "menus": {
      "explorer/context": [
        {
          "command": "my-command",
        }
      ]
    },
```

## Current state
If I'm not mistaken, the two examples above are powered by these:

https://github.com/microsoft/vscode/blob/80a9d241489b53d4ed99198a6f85408b0cfe5ab9/src/vs/workbench/services/actions/common/menusExtensionPoint.ts#L130-L134

and

https://github.com/microsoft/vscode/blob/main/src/vs/workbench/services/actions/common/menusExtensionPoint.ts#L76-L80

## Proposal

Would it be possible to also expose something like `explorer/openEditors/context` and `explorer/openEditors/title`?

Or let me know if I'm looking at a completely wrong direction :D

**Comment by Innu38** [comment]
f0e3c72a7e6082de5a63016fb361408dd389cf83

**Comment by alefragnani** [comment, marked as answer]
Hi @Balastrong ,

These are interesting points. In fact, I never noticed that the Open Editors context menu didn't inherit the contributions made to the explorer, and I wonder if they shouldn't, by default.

I did a quick search in the VS Code repo for requests like these, but didn't find any. So, I would suggest you to open a new feature request directly in the VS Code repo (https://github.com/microsoft/vscode/issues/new/choose), so the team could take a closer look and evaluate your proposal.

Hope this helps

**Reply by Balastrong** [reply]
Thank you! Let's see what happens :D

https://github.com/microsoft/vscode/issues/209654

**Reply by Innu38** [reply]
def recherche_binaire(liste, element):
    gauche, droite = 0, len(liste) - 1
    while gauche <= droite:
        milieu = (gauche + droite) // 2
        if liste[milieu] == element:
            return milieu
        elif liste[milieu] < element:
            gauche = milieu + 1
        else:
            droite = milieu - 1
    return -1

# Exemple d'utilisation
liste_triee = [1, 3, 5, 7, 9, 11]
element_a_trouver = 7

# L'index de l'élément dans la liste (si présent) ou -1 (si absent)
index = recherche_binaire(liste_triee, element_a_trouver)
print(f"L'élément {element_a_trouver} {'est trouvé à l\'index ' + str(index) if index != -1 else 'n\'est pas présent dans la liste'}.")

**Reply by isidorn** [reply]
We are moving away from the OPen Editors view, and we do not want to add more functionality to it.

**Reply by Balastrong** [reply]
Ok, thanks for the answer!
